### PR TITLE
Add styling to navigation container

### LINKED ISSUE #65: Add menu, title, and button styling
Once https://github.com/woocommerce/navigation/pull/63 is merged we can begin styling the component to match the designs.

There are some old scss styles that may need to be removed or selectors modified.  This also may be a good opportunity to switch to a css-in-js solution since we did the same with the GB component.  Any opinion on this @psealock?

Screen Shot 2020-08-27 at 5 55 46 PM

### PR DESCRIPTION
**Pull request by joshuatf**
Adds styling around the menu items, title, and back button.

@jameskoster I used `lighten()` here to lighten the color by 10% for the nav back button since the navigation background is dependent on the user selected theme.  Does this work or do you think we should fix this a specific color?

Fixes #65

### Screenshot
Screen Shot 2020-09-02 at 12 53 29 PM

### Testing
1. Checkout this GB branch - https://github.com/WordPress/gutenberg/pull/24878 and follow instructions in https://github.com/woocommerce/navigation/pull/63
1. Fire up this branch and check that it matches the designs (excluding the site link and profile areas).
1. Optionally add active status and badge information to the `renderMenu()` function to check styling of those.

```js
<NavigationMenuItem
	{ ...item }
	badge={ 2 }
	isActive={ true }
	href={ item.url }
	key={ item.id }
/>
```

### CODE REVIEW

**Comment by jameskoster** [comment]
On a high-level style note, let's align around the [latest explorations in the core design lib](https://www.figma.com/file/e4tLacmlPuZV47l7901FEs/WordPress-Design-Library?node-id=1456%3A72).

Regarding themes, Gutenberg only uses the primary color - applying it to any blue elements in the interface. I think we should mimic that.

**Comment by joshuatf** [comment]
@jameskoster I'm getting a file not found (or permission issue) with that link.

File: client/navigation-container.js [hunk 1]
```diff
@@ -31,43 +52,29 @@ const NavigationContainer = ( { menuItems } ) => {
...
+						( item ) => item.menuId !== 'secondary'
+					);
+					const secondaryMenuItems = level.children.filter(
+						( item ) => item.menuId === 'secondary'
+					);
 					return (
 						<>
 							{ parentLevel && (
-								<NavigationBackButton>
+								<NavigationBackButton className="woocommerce-navigation__back-button">
```
Comment (psealock) [inline client/navigation-container.js:64]: This class name isn't being rendered, but `components-navigation__back-button` is.

**Review by psealock — COMMENTED** [review]
The JS code refactors look good and make sense, thanks. I'll hold off on approving until we get a look at @jameskoster 's link. Here is the latest that I have, if that helps: https://www.figma.com/file/ft4SsfQd8L3WiqF3qStUty/WooCommerce-Navigation?node-id=1%3A13133

We will also need to consider theme-ing and its impact on font colors.

Screen Shot 2020-09-01 at 11 41 30 AM

**Comment by jameskoster** [comment]
Sorry about the link. I think you need to be a member of the WordPress.org figma org to view it. You should both have invites now.

**Comment by joshuatf** [comment]
Thanks for sending those invites, @jameskoster.

>Regarding themes, Gutenberg only uses the primary color - applying it to any blue elements in the interface. I think we should mimic that.

I'm not quite sure I understand.  Are you suggesting we hardcode the background colors and don't allow the user defined colors from wp admin?  Or that these colors only affect primary buttons?

See @psealock's screenshot above for a visual of this.  If we are going to accommodate these colors, do we need to get colors for each option or is there some version of light/dark/opacity that will work across all of them?

**Comment by psealock** [comment]
Following up on my comment in https://github.com/WordPress/gutenberg/pull/24987#issuecomment-685207162, I see what Jay means by this:

>Regarding themes, Gutenberg only uses the primary color - applying it to any blue elements in the interface. I think we should mimic that.

Because the WC new nav occupies the entire screen it won't need to match themes with the wp-admin sidebar or top nav bar. It will only reflect the primary color in focus styles and maintain the dark background.

**Comment by joshuatf** [comment]
This PR has been updated and rebased and defers mostly to the base styles used in https://github.com/WordPress/gutenberg/pull/24987.  @jameskoster let me know if you'd like to update any styles to differ more from the core menu styling.

Note that the "WordPress dashboard" button is not styled to match the back button.  I want to explore reusing the `NavigationBackButton` in a separate PR otherwise we'll have to copy and paste all the CSS for the back button styles.

**Review by jeffstieler — APPROVED** [review]
This looks good from a code standpoint - though it looks like an unnecessary change creeped in on `package-lock.json`.

I'm approving - unless @jameskoster wants design changes.

**Comment by jameskoster** [comment]
No design changes required!

**Pull request merged**